packages/protvista-track/src/protvista-track.js

```diff
@@ -176,86 +185,53 @@ class ProtvistaTrack extends ProtvistaZoomable {
       .attr("fill", f => this._getFeatureColor(f.feature))
       .attr("stroke", f => this._getFeatureColor(f.feature))
       .on("mouseover", f => {
-        var self = this;
-        var e = d3Event;
-
-        if (this._tooltipEvent === "mouseover") {
-          window.setTimeout(function() {
-            self.createTooltip(e, f);
-          }, 50);
-        }
         this.dispatchEvent(
-          new CustomEvent("change", {
-            detail: {
-              highlightend: f.end,
-              highlightstart: f.start,
-              highlight: `${f.start}:${f.end}`
-            },
-            bubbles: true,
-            cancelable: true
-          })
+          this._createEvent(
+            "mouseover",
+            f,
+            this._highlightEvent === "onmouseover"
+          )
         );
       })
-      .on("mouseout", () => {
-        var self = this;
-
-        if (this._tooltipEvent === "mouseover") {
-          window.setTimeout(function() {
-            self.removeAllTooltips();
-          }, 50);
-        }
+      .on("mouseout", f => {
         this.dispatchEvent(
-          new CustomEvent("change", {
-            detail: {
-              highlightend: null,
-              highlightstart: null,
-              highlight: null
-            },
-            bubbles: true,
-            cancelable: true
-          })
+          this._createEvent(
+            "mouseout",
+            null,
+            this._highlightEvent === "onmouseover"
+          )
         );
       })
-      .on("click", d => {
-        if (this._tooltipEvent === "click") {
-          this.createTooltip(d3Event, d, true);
-        }
+      .on("click", f => {
+        this.dispatchEvent(
+          this._createEvent("click", f, this._highlightEvent === "onclick")
+        );
       });
   }
 
-  createTooltip(e, d, closeable = false) {
-    if (!d.feature || !d.feature.tooltipContent) {
-      return;
-    }
-    this.removeAllTooltips();
-    const tooltip = document.createElement("protvista-tooltip");
-    tooltip.top = e.clientY + 3;
-    tooltip.left = e.clientX + 2;
-    tooltip.title = `${d.feature.type} ${d.start}-${d.end}`;
-    tooltip.closeable = closeable;
-    // Passing the content as a property as it can contain HTML
-    tooltip.content = d.feature.tooltipContent;
-    this.appendChild(tooltip);
-
-    const parentWidth = this.svg._groups[0][0].clientWidth;
-    const tooltipPosition = select(tooltip)
-      .node()
-      .getBoundingClientRect();
-
-    if (tooltipPosition.width + tooltipPosition.x > parentWidth) {
-      this.removeChild(tooltip);
-
-      tooltip.left = parentWidth - tooltipPosition.width;
-      tooltip.mirror = "H";
-
-      this.appendChild(tooltip);
+  _getCoords() {
+    if (!event) {
```

Comment:
event is imported in line1 as `d3Event`, shouldn't that be the variable to use?

Reply:
Good point